application/src/main/java/dev/stratospheric/todoapp/config/AwsConfig.java

```diff
@@ -15,21 +15,23 @@ public class AwsConfig {
 
   @Bean
   @ConditionalOnProperty(prefix = "custom", name = "use-cognito-as-identity-provider", havingValue = "true")
-  public AWSCognitoIdentityProvider awsCognitoIdentityProvider(@Value("${cloud.aws.region.static}") String region,
-                                                               AWSCredentialsProvider awsCredentialsProvider) {
+  public AWSCognitoIdentityProvider awsCognitoIdentityProvider(
+    RegionProvider regionProvider,
+    AWSCredentialsProvider awsCredentialsProvider) {
     return AWSCognitoIdentityProviderAsyncClientBuilder.standard()
       .withCredentials(awsCredentialsProvider)
-      .withRegion(region)
+      .withRegion(regionProvider.getRegion().getName())
       .build();
   }
 
   @Bean
   @ConditionalOnProperty(prefix = "custom", name = "provide-dynamodb-via-aws", havingValue = "true")
-  public AmazonDynamoDB amazonDynamoDB(@Value("${cloud.aws.region.static}") String region,
-                                       AWSCredentialsProvider awsCredentialsProvider) {
+  public AmazonDynamoDB amazonDynamoDB(
+    RegionProvider regionProvider,
```

Comment:
better to inject the `RegionProvider` so that we're decoupled from any potential configuration changes of properties within Spring Cloud AWS.